src/main/java/use_cases/add_edit_category_use_case/CategoryUCI.java

```diff
@@ -14,10 +11,16 @@
  * Implements CategoryIB.
  * Creates CategoryOD objects related to specific use case fail/success conditions.
  */
-
 public class CategoryUCI implements CategoryIB {
     private final CategoryOB categoryOB;
+    private CategoryID categoryID;
+    private Category selectedCategory;
+    private double valueDouble;
     private final CategoryFactory categoryFactory;
```

Comment:
I don't really mind, but wouldn't it be a little confusing since this use case is only ever going to be dealing with Category entity?

Reply:
Yes but we want the code to be as general as possible. And since the only method you're gonna be using from the factory is a method contained in the abstract class, you can just keep the type signature for the factory abstract too

Reply:
Why did you switch it back to CategoryFactory?

Reply:
MonthObjectFactory isn't public so I'm not sure how I can declare it.

Reply:
> MonthObjectFactory isn't public so I'm not sure how I can declare it.

I believe the MonthObjectFactory is a public class, so I could just declare it like how you declare any other class object. From what I understand, it should work somewhat like this:

MonthObjectFactoryInputData factoryInput = new CategoryCreaterInputData(input goes here);
this.categoryFactory = new MonthObjectFactory(factoryInput);

Reply:
They are actually abstract so they are not public! I apologize, it looks like Ari fixed this in his Intellij warning branch. https://github.com/CSC207-2023Y-UofT/course-project-DoorKeepers/pull/59/commits/90719a0e91d12b4fc309817448c2a4d340006a88

Reply:
You could just add the public modifiers in your version to make sure it works.

Reply:
I've made them public but that still doesn't change the fact that I can't instantiate an abstract class. 
So: `this.categoryFactory = new `MonthObjectFactory(factoryInput);` won't work?

Reply:
> I've made them public but that still doesn't change the fact that I can't instantiate an abstract class. So: `this.categoryFactory = new `MonthObjectFactory(factoryInput);` won't work?

Ah, my bad. You should declare the private attribute like this:
private final MonthObjectFactory categoryFactory;

then do:
this.categoryFactory = new CategoryFactory(factoryInput);

Reply:
You can declare it as the CategoryCreatorInputData and only cast it at the return statement

Reply:
> I've made them public but that still doesn't change the fact that I can't instantiate an abstract class.
> So: `this.categoryFactory = new `MonthObjectFactory(factoryInput);` won't work?

Also just to mention here, the reason why you can’t make an instance of it is because it is abstract, and java prevents you from instantiating an abstract class! So casting in the return like Ari said is the way you are supposed to do it.